test/test_torch.py

```diff
@@ -6977,15 +6977,64 @@ def compare_strides(s1, s2, div):
 
     @skipMeta
     @dtypes(*torch.testing.get_all_dtypes())
-    def test_dlpack_conversion(self, device, dtype):
-        # DLpack does not explicitly support bool
+    def test_dlpack_capsule_conversion(self, device, dtype):
+        # DLpack does not explicitly support bool (xref dmlc/dlpack#75)
         # It does it through uint8 type
-        if dtype is torch.bool:
+        if dtype is torch.bool or 'xla' in device:
             return
         x = make_tensor((5,), device, dtype, low=-9, high=9)
         z = from_dlpack(to_dlpack(x))
         self.assertEqual(z, x)
 
+    @skipMeta
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_protocol_conversion(self, device, dtype):
+        if dtype is torch.bool or 'xla' in device:
+            return
+        x = make_tensor((5,), device, dtype, low=-9, high=9)
+        z = from_dlpack(x)
+        self.assertEqual(z, x)
+
+    @skipMeta
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_conversion_with_streams(self, device, dtype):
+        if dtype is torch.bool or 'xla' in device:
+            return
+        # Create a stream where the tensor will reside
+        if device == 'cuda':
+            x = make_tensor((5,), device, dtype, low=-9, high=9)
+            stream = torch.cuda.Stream()
```

Comment:
Do you need a synchronize call here to ensure `x` has been populated on a CUDA device before its accessed on a different stream?

Reply:
Here we want to test how the dlpack interface does this synchronization, so we avoid doing it here.

Reply:
I added a comment to clarify this